### PR TITLE
Fix migrate

### PR DESCRIPTION
**Pull request by july2993**
1. Fix wrong length(less then the finally file write).
2. Fix losing snapshot role in root.json

### CODE REVIEW

File: tools/migrate/main.go [hunk 1]
```diff
@@ -309,14 +305,24 @@ func migrate(srcDir, dstDir string) error {
...
+
+	hash, _, err := hashManifest(signedManifests[v1manifest.ManifestTypeSnapshot])
 	if err != nil {
 		return errors.Trace(err)
 	}
 	timestamp.Meta = map[string]v1manifest.FileHash{
 		v1manifest.ManifestFilenameSnapshot: {
 			Hashes: hash,
-			Length: n,
+			Length: limitLength,
```
Comment (lonng) [inline tools/migrate/main.go:325]: Is it correct? I think the length should be the real file length of snapshot.json.

Comment (july2993) [reply tools/migrate/main.go:325]: >= the length of snapshot.sjon is fine.
even zero should be ok which means no limit.

Comment (nrc) [reply tools/migrate/main.go:325]: 0 is not ok (might permit DOS-style attacks), but the same or greater (within reason) is ok.

Comment (july2993) [reply tools/migrate/main.go:325]: when 0, client should use a limit size at the client side to protest this attack, then I think it's ok?
seems I read this in the TUF spec before, but I am not sure.

Comment (nrc) [reply tools/migrate/main.go:325]: I think it is true, but we have not implemented this

**Pull request merged**